### PR TITLE
Protect sidekiq

### LINKED ISSUE #106: Restrict access to Sidekiq admin interface to admin users only
Blocked by #37 

Once we can track a user's group membership, we can restrict access to the Sidekiq interface based on UMG membership. The current scholarsphere admin group is `up.dlt.scholarsphere-admin` so any member of that group should have admin privileges.

### PR DESCRIPTION
**Pull request by whereismyjetpack**
Fixes #106

Restricts sidekiq admin UI to admin users and adds Redis to the docker build.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -29,6 +29,8 @@ class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 6.0
 
+    config.admin_group = ENV.fetch('ADMIN_GROUP', 'cn=umg/up.libraries.scholarsphere.service-team,dc=psu,dc=edu')
```
Comment (awead) [inline config/application.rb]: this should be `cn=umg/up.dlt.scholarsphere-admin,dc=psu,dc=edu`

File: config/initializers/scholarsphere_admin.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class ScholarsphereAdmin
+  def self.matches?(request)
+    current_user = request.env['warden'].user
+    return false if current_user.blank?
+
+    current_user.groups.any? { |g| g.name == Scholarsphere::Application.config.admin_group }
```
Comment (awead) [inline config/initializers/scholarsphere_admin.rb]: `current_user.admin?`

Then, create a method on User:

``` ruby
def admin?
  groups.include? Scholarsphere::Application.config.admin_group
end
```

File: config/samples/application.yml [hunk 3]
```diff
@@ -38,6 +38,10 @@
 #
 
 
+## ADMIN UMG
+## protects the admin namespace 
+# ADMIN_GROUP: "cn=umg/up.libraries.scholarsphere.service-team,dc=psu,dc=edu"
```
Comment (awead) [inline config/samples/application.yml]: change to cn=umg/up.dlt.scholarsphere-admin,dc=psu,dc=edu

File: config/application.rb (hunk 1 above)
Comment (whereismyjetpack) [reply config/application.rb]: fixed!

**Comment by whereismyjetpack** [comment]
@awead this should be good to go now

File: config/initializers/scholarsphere_admin.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class ScholarsphereAdmin
+  def self.matches?(request)
+    current_user = request.env['warden'].user
+    return false if current_user.blank?
+
+    current_user.admin?
+  end
```
Comment (awead) [inline config/initializers/scholarsphere_admin.rb]: @whereismyjetpack this isn't really tested. We're testing `admin?`, which is good, but we ought to have a test the exercises this block. A request spec might work here. Hit the endpoint as an admin and verify it's successful, then hit it as a non-admin and it should be 404, 403, or?

File: app/models/user.rb [hunk 5]
```diff
@@ -45,6 +45,10 @@ def self.from_omniauth(auth)
     user
   end
 
+  def admin?
+    groups.any? { |g| g.name == Scholarsphere::Application.config.admin_group }
```
Comment (awead) [inline app/models/user.rb]: @whereismyjetpack do you find

``` ruby
groups(&:name).include? Scholarsphere::Application.config.admin_group
```
easier to read?

File: config/initializers/scholarsphere_admin.rb (hunk 4 above)
Comment (whereismyjetpack) [reply config/initializers/scholarsphere_admin.rb]: Excellent! I'll certainly add another spec

Comment (whereismyjetpack) [reply config/initializers/scholarsphere_admin.rb]: I added some request specs that exercise 3 cases on this endpoint. logged out, logged in, and logged in as admin.

File: spec/request/sidekiq_admin_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Sidekiq admin', type: :request do
+  let(:user) { create(:user) }
+  let(:admin_user) { build_stubbed :user,
+                                   given_name: 'Joe',
+                                   surname: 'Developer',
+                                   groups: [Group.new(name: 'cn=umg/up.dlt.scholarsphere-admin,dc=psu,dc=edu')] }
```
Comment (awead) [inline spec/request/sidekiq_admin_spec.rb]: @whereismyjetpack how about a trait here. `let(:user) { build(:user, :is_admin) }` Then you define a trait in `spec/factories/users.rb` that adds the admin group. The given and surname are inconsequential and get generated by the factory.

Comment (whereismyjetpack) [reply spec/request/sidekiq_admin_spec.rb]: Thanks for that, I wasn't sure how best to handle that. I knew pasting that umg everhwere smelled bad, just couldn't tell where the smell was coming from.

**Pull request merged**